File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Horse_Rider.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Horse;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Horse_Rider implements Listener {
    @EventHandler
    public void onDamage(final EntityDamageByEntityEvent entityDamageByEntityEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "horse_rider"));
        if (entityDamageByEntityEvent.getDamager() instanceof Player && !(entityDamageByEntityEvent.getEntity() instanceof Player)) {
            try {
                final Player owner = (Player) entityDamageByEntityEvent.getDamager();
                ItemStack[] armorContents= owner.getInventory().getArmorContents();
                for (int length = armorContents.length, i = 0; i < length; ++i) {
                    final ItemStack itemStack = armorContents[i];
                    if (itemStack != null && itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && entityDamageByEntityEvent.getEntity() instanceof Horse) {
                        entityDamageByEntityEvent.setCancelled(true);
                        entityDamageByEntityEvent.setDamage(0.0);
                        final Horse horse = (Horse) entityDamageByEntityEvent.getEntity();
                        if (horse.isTamed()) {
                            return;
                        }
                        horse.setTamed(true);
                        horse.setOwner(owner);
                        horse.getInventory().setSaddle(new ItemStack(Material.SADDLE));
                    }
                }
            } catch (Exception ex) {
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantments/Horse_Rider.kt
package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.entity.Horse
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.inventory.ItemStack
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Horse_Rider : Listener {
    @EventHandler
    fun onDamage(entityDamageByEntityEvent: EntityDamageByEntityEvent) {
        val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "horse_rider"))
        if (entityDamageByEntityEvent.damager is Player && entityDamageByEntityEvent.entity !is Player) {
            try {
                val owner = entityDamageByEntityEvent.damager as Player
                val armorContents = owner.inventory.armorContents
                val length = armorContents.size
                var i = 0
                while (i < length) {
                    val itemStack = armorContents[i]
                    if (itemStack != null && itemStack.hasItemMeta() && itemStack.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " I") && entityDamageByEntityEvent.entity is Horse) {
                        entityDamageByEntityEvent.isCancelled = true
                        entityDamageByEntityEvent.damage = 0.0
                        val horse = entityDamageByEntityEvent.entity as Horse
                        if (horse.isTamed) {
                            return
                        }
                        horse.isTamed = true
                        horse.owner = owner
                        horse.inventory.saddle = ItemStack(Material.SADDLE)
                    }
                    ++i
                }
            } catch (ex: Exception) {
            }

        }
    }
}
